### ISSUE TITLE
Passing custom annotations to backup pods

### ISSUE THREAD

**Issue by laukaichung** [reactive-tech/kubegres#15]
This is a ticket based on https://github.com/reactive-tech/kubegres/issues/14
I would like to pass custom annotations to backup pods so that third party apps can access backup pods like they do with the master and replica pods.

**Comment by alex-arica** [comment]
I made the changes in GIT main and you can install the version 1.5 from the "kubegres.yaml" in main, as follows:

```
kubectl apply -f https://raw.githubusercontent.com/reactive-tech/kubegres/main/kubegres.yaml
```

From this version, the Pods created by the backup CronJob contain the custom annotations and the custom environment variables defined in the YAML of "kind: Kubegres".

Note that in the default backup script, the env variable "$KUBEPOSTGRES_RESOURCE_NAME" had a typo, and I renamed it to "$KUBEGRES_RESOURCE_NAME". That variable contains the name given to the PostgreSql cluster in the deployment YAML of "kind: Kubegres". It is used by default to name the backup files.

Once you are happy with your tests, please let me know and I will then update the acceptance tests and create a new release.

**Comment by laukaichung** [comment]
Thank you, I'll get back to you in a few hours~

**Comment by laukaichung** [comment]
Working fine in both Hetzner cluster and Kind. The feature should be complete! 
I've made a cluster example that is able to upload pg_dump results to S3 and downloading the latest for primary initialization . But I've decided to use a custom Docker image with AWS cli and curl installed, since the scripts do not have the permission to run `apt install`.

https://github.com/laukaichung/kubegres-deploy

**Comment by alex-arica** [comment]
Thank you for your feedback and your help! 

I will add a link from the official documentation to your repository so that other users of Kubegres can benefit from your examples about how to integrate with Bank-Vaults.

I will let you know once a tag & release are available for version 1.5

You made great suggestions to improve Kubegres. If you have additional ones please don't hesitate to create a ticket.

**Comment by alex-arica** [comment]
I created a new tag/release 1.5 here:
https://github.com/reactive-tech/kubegres/releases

To install Kubegres 1.5 :
```
kubectl apply -f https://raw.githubusercontent.com/reactive-tech/kubegres/v1.5/kubegres.yaml
```

I updated the documentation pages, as follows:
- new version 1.5 added in: [Getting started page](https://www.kubegres.io/doc/getting-started.html)
- a link to your excellent repository in: [Override the default configurations](https://www.kubegres.io/doc/override-default-configs.html)

Please close this ticket if you are happy with the changes.

Thank you for your suggestions and your help!
